nabavka/src/main/java/raf/si/racunovodstvo/nabavka/controllers/KalkulacijaController.java

```diff
@@ -0,0 +1,76 @@
+package raf.si.racunovodstvo.nabavka.controllers;
+
+import io.swagger.v3.oas.annotations.security.SecurityRequirement;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import raf.si.racunovodstvo.nabavka.model.Kalkulacija;
+import raf.si.racunovodstvo.nabavka.requests.KalkulacijaRequest;
+import raf.si.racunovodstvo.nabavka.services.impl.KalkulacijaService;
+import raf.si.racunovodstvo.nabavka.utils.ApiUtil;
+import raf.si.racunovodstvo.nabavka.utils.SearchUtil;
+
+import javax.persistence.EntityNotFoundException;
+import javax.validation.Valid;
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+import java.util.Optional;
+
+@CrossOrigin
+@RestController
+@SecurityRequirement(name = "bearerAuth")
+@RequestMapping("/api/kalkulacije")
+public class KalkulacijaController {
+
+    private final KalkulacijaService kalkulacijaService;
+
+    private final SearchUtil<Kalkulacija> searchUtil;
+
+    public KalkulacijaController(KalkulacijaService kalkulacijaService) {
+        this.kalkulacijaService = kalkulacijaService;
+        this.searchUtil = new SearchUtil<>();
+    }
+
+    @GetMapping(produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> searchKalkulacija(
+        @RequestParam(name = "search", required = false, defaultValue = "") String search,
+        @RequestParam(defaultValue = ApiUtil.DEFAULT_PAGE) @Min(ApiUtil.MIN_PAGE) Integer page,
+        @RequestParam(defaultValue = ApiUtil.DEFAULT_SIZE) @Min(ApiUtil.MIN_SIZE) @Max(ApiUtil.MAX_SIZE) Integer size,
+        @RequestParam(defaultValue = "brojKalkulacije") String[] sort
+    ) {
+        Pageable pageSort = ApiUtil.resolveSortingAndPagination(page, size, sort);
+        Specification<Kalkulacija> spec = this.searchUtil.getSpec(search);
+        return ResponseEntity.ok(this.kalkulacijaService.findAll(spec, pageSort));
+    }
```

Comment:
Mislim da nije pokriven slucaj kada je search prazan.

Reply:
Nije li logicnije da se prazan search hendluje unutar SearchUtila nekako? Malo je cudno da za svaku rutu gde ga koristimo moramo zasebno to da radimo.

Reply:
Sto se mene tice mozes da probas da handle-ujes unutar SearchUtil-a, sve dok ti ta ruta na kraju radi. Trenutno ako ti neko posalje prazan string za search mislim da ce dobiti internal server error, sto znaci da ti ne radi taj endpoint onako kako je ocekivano.